Cannot build on Fedora 34

I've been unable to get Sandstorm to build on Fedora 34.  I've created [a minimal Dockerfile to demonstrate the failure](https://gist.github.com/garrison/f025090ce914641f0d1e492612dcbd4e#file-dockerfile-fedora), as I mentioned in #3564.  If I [increase the log output](https://groups.google.com/g/sandstorm-dev/c/ARrmLkmWJcw/m/Em91iRE0DAAJ) and build, everything is successful until

```
✘ link: sandstorm/run-bundle.o
✘ install: sandstorm/sandstorm.ekam-manifest
✘ link: sandstorm/sandstorm-http-bridge.o
✘ link: sandstorm/test-app/test-app.o
    /usr/bin/ld: cannot find -lz
make: *** [Makefile:268: tmp/.ekam-run] Error 1
```

despite both `zlib` and `zlib-devel` being installed.

--- a/docs/install.md
+++ b/docs/install.md
@@ -131,10 +131,13 @@ Please install the following:
 * Linux x86_64, with kernel version 3.13 or later
 * C and C++ standard libraries and headers
 * GNU Make
+* GNU diffutils
+* `g++`
 * `libcap` with headers
 * `xz`
 * `zip`
 * `unzip`
+* `which`
 * `flex`
 * `bison`
 * `strace`
@@ -154,12 +157,12 @@ On Debian or Ubuntu, you should be able to get all these with:
         golang-go cmake strace flex bison locales
     curl https://install.meteor.com/?release=1.8.2 | sh
 
-On Fedora 27 you should be able to get them with (as root):
+On Fedora 34 you should be able to get them with:
 
-    dnf install make libcap-devel libstdc++-devel libstdc++-static \
-       glibc-headers glibc-static glibc-locale-source xz zip \
-       unzip strace curl discount git python2 zlib-devel \
-       golang cmake strace flex bison
+    sudo dnf install make libcap-devel libstdc++-devel libstdc++-static \
+       glibc-headers glibc-static glibc-locale-source gcc-c++ xz zip \
+       unzip strace curl discount git python zlib-devel zlib-static \
+       golang cmake strace flex bison which diffutils
     curl https://install.meteor.com/?release=1.8.2 | sh
 
 If you have trouble getting the build to work on your distro, we recommend trying in a virtual

